fakestorage/upload.go

```diff
@@ -146,9 +146,15 @@ func (s *Server) multipartUpload(bucketName string, w http.ResponseWriter, r *ht
 		http.Error(w, err.Error(), http.StatusInternalServerError)
 		return
 	}
+
+	objName := r.URL.Query().Get("name")
+	if objName == "" && metadata != nil {
```

Comment:
It's safe to assume metadata is not nil here.

```suggestion
	if objName == "" {
```

Reply:
Never mind, I take that back.

I'll add a test later to ensure we handle this properly x)